invoicer_project/static/js/registration_request.js

```diff
@@ -0,0 +1,269 @@
+        emailField.setAttribute("errorText", errorsObject['emailValidationResult']);
+    }
+    if (errorsObject['passwordValidationResult']) {
+        passwordField.setAttribute("error", "true");
+        passwordField.setAttribute("errorText", errorsObject['passwordValidationResult']);
+    }
+    if (errorsObject['repeatPasswordValidationResult']) {
+        repeatPasswordField.setAttribute("error", "true");
+        repeatPasswordField.setAttribute("errorText", errorsObject['repeatPasswordValidationResult']);
+    }
+    if (errorsObject['doPasswordsMatch']) {
+        passwordField.removeAttribute("errorText");
+        repeatPasswordField.removeAttribute("errorText");
+        passwordField.setAttribute("error", "true");
+        repeatPasswordField.setAttribute("error", "true");
+        passwordField.setAttribute("errorText", errorsObject['doPasswordsMatch']);
+    }
+}
+
+nameField.addEventListener('input', () => {
+    nameField.removeAttribute("errorText");
+    nameField.removeAttribute("error");
+});
+
+surnameField.addEventListener('input', () => {
+    surnameField.removeAttribute("errorText");
+    surnameField.removeAttribute("error");
+});
+
+passwordField.addEventListener('input', () => {
+    passwordField.removeAttribute("errorText");
+    passwordField.removeAttribute("error");
+    repeatPasswordField.removeAttribute("errorText");
+    repeatPasswordField.removeAttribute("error");
+});
+
+repeatPasswordField.addEventListener('input', () => {
+    repeatPasswordField.removeAttribute("errorText");
+    repeatPasswordField.removeAttribute("error");
+    passwordField.removeAttribute("errorText");
+    passwordField.removeAttribute("error");
+});
+
+emailField.addEventListener('input', () => {
+    emailField.removeAttribute("errorText");
+    emailField.removeAttribute("error");
+});
+
+async function actionAfterRegisterRequest(registerStatusCode, dataToSend) {
+    if (registerStatusCode === 200) {
+        const tokenObtainStatusCode = await obtainAndSaveTokens(host + '/user/login/', dataToSend);
+        if (tokenObtainStatusCode === 200) {
+            window.location.replace(host + '/clients/home/');
+        } else {
+            showBackEndErrorsAtFE(tokenObtainStatusCode);
+        }
+    } else {
+        emailField.setAttribute("error", "true");
+        passwordField.setAttribute("error", "true");
+        repeatPasswordField.setAttribute("error", "true");
+        if (registerStatusCode === 400) {
+            passwordField.setAttribute("errorText", "Invalid credentianls!");
+        } else if (registerStatusCode === 409) {
+            passwordField.removeAttribute("error");
+            repeatPasswordField.removeAttribute("error");
+            emailField.setAttribute("errorText", "User with such email already exists!");
+        } else {
+            passwordField.setAttribute("errorText", "Unknown server error");
+        }
+    }
+}
+
+function showBackEndErrorsAtFE(status) {
```

Comment:
What is FE?

Reply:
done, corrected to 'showBackEndErrorsAtFrontEnd'